src/com/google/enterprise/adaptor/DocumentContentTransformer.java

```diff
@@ -0,0 +1,58 @@
+package com.google.enterprise.adaptor;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * The document content transformer can modify the content of a document.
+ *
+ * @author Dominik Weidenfeld (dominik.weidenfeld@twt.de)
+ */
+public class DocumentContentTransformer extends OutputStream {
+
+  protected final Map<String, String> config;
+  protected final Metadata metadata;
+  protected final String contentType;
+  private OutputStream originalStream;
+
+  public DocumentContentTransformer(final Map<String, String> config,
+                                    final OutputStream originalStream, final String contentType,
+                                    final Metadata metadata) {
+    // config and metadata cannot be null
```

Comment:
there are already all necessary null checks. The other ones are not mandatory (see new javadoc comments)